src/mlpack/methods/ann/layer/sigmoid_cross_entropy_error.hpp

```diff
@@ -0,0 +1,117 @@
+/**
+ * @file sigmoid_cross_entropy_error.hpp
+ * @author Kris Singh
+ *
+ * Definition of the cross-entropy with logit performance function.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_SIGMOID_CROSS_ENTROPY_ERROR_HPP
+#define MLPACK_METHODS_ANN_LAYER_SIGMOID_CROSS_ENTROPY_ERROR_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+/**
+ * The SigmoidCrossEntropyError performance function measures the network's
+ * performance according to the cross-entropy function between the input and
+ * target distributions. This function calculates the cross entropy
+ * given the real values instead of providing the sigmoid activations.
+ * The function is much more numerically stable and uses this
+ * equivalent formulation:
+ * \f$max(x, 0) - x * z + \log(1 +  e^{-|x|})\f$
+ * where x = input and z = target.
+ *
+ * For more information, see the following paper.
+ *
+ * @code
+ * @article{1702.05659,
+ *   title={On Loss Functions for Deep Neural Networks in Classification},
+ *   author={Katarzyna Janocha, Wojciech Marian Czarnecki},
+ *   eprint={arXiv:1702.05659},
+ *   year={2017}
```

Comment:
Can you add ``url       = {http://arxiv.org/abs/1702.05659}`` and ``journal   = {CoRR},``?